src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java

```diff
@@ -381,33 +381,21 @@ public void setCredentialsId(String credentialsId) {
     }
 
     public int getContainerCap() {
-        return containerCap;
+        return containerCap != null ? containerCap : Integer.MAX_VALUE;
     }
 
     @DataBoundSetter
     public void setContainerCapStr(String containerCapStr) {
-        if (containerCapStr.equals("")) {
-            setContainerCap(Integer.MAX_VALUE);
-        } else {
-            setContainerCap(Integer.parseInt(containerCapStr));
-        }
+        setContainerCap(containerCapStr.equals("") ? null : Integer.parseInt(containerCapStr));
     }
 
-    @DataBoundSetter
-    public void setContainerCap(int containerCap) {
-        if (containerCap < 0) {
-            this.containerCap = Integer.MAX_VALUE;
-        } else {
-            this.containerCap = containerCap;
-        }
+    public void setContainerCap(Integer containerCap) {
+        this.containerCap = (containerCap != null && containerCap >= 0) ? containerCap : null;
```

Comment:
Is 0 a valid value?

Reply:
Yes, means that the cloud is effectively disabled.